staging/src/k8s.io/apiserver/pkg/storage/cacher.go

```diff
@@ -369,6 +369,12 @@ func (c *Cacher) Get(ctx context.Context, key string, resourceVersion string, ob
 		return err
 	}
 
+	if !c.ready.check() && getRV == 0 {
```

Comment:
also, once the cache is initialized, it seems bad to continue contending on a lock to check this forever.